src/jira_client.ts

```diff
@@ -0,0 +1,381 @@
+            body: comment
+          })
+        }),
+      toError
+    ).chain<CreateJiraCommentIssueResponse>(_ => {
+      if (_.status >= 500) {
+        return fromLeft(new Error("Jira API returns an error"));
+      }
+      if (_.status === 401) {
+        return fromLeft(new Error("Jira secrets misconfiguration"));
+      }
+      if (_.status === 400) {
+        return fromLeft(new Error("Invalid request"));
+      }
+      if (_.status !== 201) {
+        return fromLeft(new Error("Unknown status code response error"));
+      }
+      return tryCatch(() => _.json(), toError).chain(responseBody =>
+        fromEither(
+          CreateJiraCommentIssueResponse.decode(responseBody).mapLeft(errors =>
+            toError(readableReport(errors))
+          )
+        )
+      );
+    });
+
+  const getServiceJiraIssuesByStatus = (params: {
+    readonly serviceId: ServiceId;
+    readonly status: NonEmptyString;
+  }) => {
+    const bodyData: JiraIssueSearchPayload = {
+      expand: ["names"],
+      fields: ["summary", "status", "assignee", "comment"],
+      fieldsByKeys: false,
+      jql: `project = ${boardId} AND issuetype = Task AND (labels = ${JIRA_SERVICE_TAG_PREFIX}${params.serviceId} OR (labels = ${JIRA_SERVICE_TAG_PREFIX}${params.serviceId} AND labels = ${JIRA_DISABLE_LABEL})) AND status = ${params.status} ORDER BY created DESC`,
+      startAt: 0
+    };
+    return jiraIssueSearch(baseUrl, jiraEmail, token, bodyData, fetchApi);
+  };
+  const searchServiceJiraIssue = (params: {
+    readonly serviceId: ServiceId;
+  }) => {
+    const bodyData: JiraIssueSearchPayload = {
+      expand: ["names"],
+      fields: ["summary", "status", "assignee", "comment", "labels"],
+      fieldsByKeys: false,
+      // Check if is better without JIRA_SERVICE_TAG_PREFIX
+      jql: `project = ${boardId} AND issuetype = Task AND (labels = ${JIRA_SERVICE_TAG_PREFIX}${params.serviceId} AND status != ${JIRA_CONFIG.JIRA_STATUS_COMPLETE}) ORDER BY created DESC`,
+      startAt: 0
+    };
+    return jiraIssueSearch(baseUrl, jiraEmail, token, bodyData, fetchApi);
+  };
+  const applyJiraIssueTransition = (
+    issueId: NonEmptyString,
+    transitionId: NonEmptyString,
+    newComment?: NonEmptyString
+  ) => {
+    return tryCatch(() => {
+      return fetchApi(`${baseUrl}/rest/api/2/issue/${issueId}/transitions`, {
+        method: "POST",
+
+        headers: {
+          Accept: "application/json",
+          Authorization: `Basic ${Buffer.from(`${jiraEmail}:${token}`).toString(
+            "base64"
+          )}`,
+          "Content-Type": "application/json"
+        },
+
+        body: JSON.stringify({
+          ...(newComment
+            ? {
+                update: {
+                  comment: [
+                    {
+                      add: {
+                        body: newComment
+                      }
+                    }
+                  ]
+                }
+              }
+            : {}),
```

Comment:
What if we use this more compact syntax, removing the conditional ternary operator?
```suggestion
          ...fromNullable(newComment)
            .map(_ => ({
              update: {
                comment: [
                  {
                    add: {
                      body: _
                    }
                  }
                ]
              }
            }))
            .toUndefined(),
```

Reply:
Fix with 26ad9dd